### DISCUSSION TITLE
Aspire

### DISCUSSION THREAD

**Discussion by TheMagnificent11** [Depechie/OpenTelemetryGrafana#5] [General]
Now that you've updated Open Telemetry Grafana to .Net 8, do you plan to use Aspire for service orchestration instead of Docker Compose?

**Comment by Depechie** [comment]
For now with the Grafana tooling it is not yet possible.
Aspire still misses some features that we need for the correct docker setup.

But the Aspire team is aware of the missing features already and they are on the backlog to be resolved.

So as soon as that is done I will add a version that uses Aspire :)

**Reply by TheMagnificent11** [reply]
Thanks for the update.

Can you please post on here when the Aspire team fixes the issue?

**Reply by Depechie** [reply]
https://github.com/dotnet/aspire/issues/936

**Reply by TheMagnificent11** [reply]
Thanks, I subscribed to the issue

**Reply by TheMagnificent11** [reply]
Oh, that issue is closed and apparently there's a fix already.

Any idea when the next Aspire release will be?

**Reply by Depechie** [reply]
No not sure about the release cycle. Also, although I think it would fix the missing elements, I will still need to try it out when released because the original issue posted is a bit different from my scenario.

But I do hope it will be fixed :)

**Reply by TheMagnificent11** [reply]
Thanks for the quick replies

**Reply by Depechie** [reply]
First tests with preview 2 look very promising :)

**Comment by Depechie** [comment]
@TheMagnificent11 the first version that uses Aspire as orchestrator is available on the https://github.com/Depechie/OpenTelemetryGrafana/tree/aspire branch.

I'm not going to merge the 2 versions into main because there are some code adjustments for Aspire that I do not want to add to the main branch if not needed.

**Reply by TheMagnificent11** [reply]
Thanks, I'll take a look over the next few days

**Reply by TheMagnificent11** [reply]
@Depechie do you know if Microsoft will release Aspire NuGet packages for things like Grafana and Loki?

**Reply by Depechie** [reply]
No I have no idea what the roadmap for Aspire is :/
